coupler/atmos_ocean_fluxes.F90

```diff
@@ -510,10 +513,18 @@ subroutine atmos_ocean_fluxes_init(gas_fluxes, gas_fields_atm, gas_fields_ice, v
           write (outunit,*) trim(note_header), ' Processing ', gas_fluxes%num_bcs, ' gas fluxes'
     endif
 
-    ! allocate the arrays
-    allocate (gas_fluxes%bc(gas_fluxes%num_bcs))
-    allocate (gas_fields_atm%bc(gas_fields_atm%num_bcs))
-    allocate (gas_fields_ice%bc(gas_fields_ice%num_bcs))
+    if(present(use_r4_kind)) use_r4_kind_loc = use_r4_kind
```

Comment:
wondering if this should go on top before the while loop

Reply:
Not sure what you mean, I have this if for allocation here and then the while loop below has another if within it for the parts that use reals.